cmake/SpectreCheckTargetDependencies.cmake

```diff
@@ -0,0 +1,652 @@
+    list(APPEND SPECTRE_INCLUDES ${INCLUDE})
+  endforeach(INCLUDE ${_RAW_SPECTRE_INCLUDES})
+  # "Return" SPECTRE_INCLUDE by setting it in PARENT_SCOPE
+  set(SPECTRE_INCLUDES ${SPECTRE_INCLUDES} PARENT_SCOPE)
+endfunction(_extract_tpl_includes FILE_NAME)
+
+# Make the path of `HEADER_FILE` which belongs to the target `TARGET_NAME`
+# an absolute path.
+#
+# Sets the variable `ABS_HEADER_FILE` in parent scope to return the result
+function(_make_header_path_absolute TARGET_NAME HEADER_FILE)
+  get_property(
+    _INCLUDE_DIR
+    TARGET ${TARGET_NAME}
+    PROPERTY INTERFACE_INCLUDE_DIRECTORIES
+    )
+  if(IS_ABSOLUTE ${HEADER_FILE})
+    set(_ABS_HEADER_FILE ${HEADER_FILE})
+  else()
+    set(_ABS_HEADER_FILE "${_INCLUDE_DIR}/${HEADER_FILE}")
+  endif()
+  # "Return" ABS_HEADER_FILE by setting it in PARENT_SCOPE
+  set(ABS_HEADER_FILE ${_ABS_HEADER_FILE} PARENT_SCOPE)
+endfunction(_make_header_path_absolute TARGET_NAME HEADER_FILE)
+
+# Gets the dependencies for a list of includes.
+#
+# Example usage:
+#
+#   _get_deps_for_spectre_includes(
+#     ${TARGET_NAME}
+#     "${LIST_OF_ALL_TARGETS}"
+#     "${SPECTRE_INCLUDES}"
+#     "${_HEADER_FILES}"
+#     ${RECURSE_INCLUDES}
+#     )
+#
+# Arguments:
+#
+# TARGET_NAME: the target for which to check the link libraries/dependencies
+# LIST_OF_ALL_TARGETS: a list of all the targets that might be dependencies
+#                      of the target ${TARGET_NAME}
+# SPECTRE_INCLUDES_OF_FILE:
+#     a list of the includes of a file for which to get the dependencies
+# TARGET_HEADER_FILES:
+#     a list of header files for the target ${TARGET_NAME}
+# RECURSE_THROUGH_INCLUDES:
+#     if TRUE then any include in any of ${SPECTRE_INCLUDES_OF_FILE}
+#     that is also in the list ${TARGET_HEADER_FILES} has its dependencies
+#     also added.
+#
+# Since the recursion will cause an infinite loop if there are cyclic includes
+# we limit the recursion depth to 20, at which point an error message is
+# displayed that there may be cyclic dependencies
+function(_get_deps_for_spectre_includes TARGET_NAME LIST_OF_ALL_TARGETS
+    SPECTRE_INCLUDES_OF_FILE TARGET_HEADER_FILES RECURSE_THROUGH_INCLUDES)
+  # Implement guard against infinite recursion when there are cyclic
+  # includes in a library.
+  get_property(
+    GET_DEPS_FOR_SPECTRE_INCLUDES_PATH
+    GLOBAL PROPERTY GET_DEPS_FOR_SPECTRE_INCLUDES_PATH
+    )
+  list(LENGTH GET_DEPS_FOR_SPECTRE_INCLUDES_PATH RECURSE_DEPTH)
+  if(${RECURSE_DEPTH} GREATER 20)
+    message(FATAL_ERROR
+      " Found what appears to be a cyclic loop in the header files."
+      " The header path is: ${GET_DEPS_FOR_SPECTRE_INCLUDES_PATH}"
+      )
+  endif(${RECURSE_DEPTH} GREATER 20)
+
+  unset(_TARGET_DEPS)
+  # Now loop over all spectre targets to see if any of them
+  # expose these includes
+  foreach(SPECTRE_INCLUDE ${SPECTRE_INCLUDES_OF_FILE})
+    # If we are checking header files and the header file is part of
+    # this target, skip checks
```

Comment:
It's not clear which checks are being skipped, because there's no else branch ... or is this comment misplaced?